lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1697,6 +1759,14 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
 
     llvm::DINode::DIFlags Flags = llvm::DINode::FlagZero;
 
+    if (DbgTy.getType()->is<NominalOrBoundGenericNominalType>()) {
+      if (auto Parent = DbgTy.getType()->getNominalParent()) {
+        auto ParentDbgType = DebugTypeInfo::getFromTypeInfo(
+            Parent, IGM.getTypeInfoForUnlowered(Parent), IGM);
+        Scope = getOrCreateType(ParentDbgType);
```

Comment:
Comment why scope is only set for nominals?

Reply:
Turns out this is not quite right either because of private discriminators... I'll deal with the nesting issue in a separate patch.